src/init.cpp

```diff
@@ -2484,8 +2484,18 @@ bool AppInitMain(InitInterfaces& interfaces)
 
         const UniValue b = blockToJSON(block, tip, pblockindex, true, 2);
 
+        rust::cxxbridge1::Vec<PoolCreationHeight> pools;
+        pcustomcsview->ForEachPoolPair(
+            [&](DCT_ID const &id, CPoolPair pool) {
+                const auto token = pcustomcsview->GetToken(id);
+                if (token) {
+                    pools.push_back({id.v, pool.creationHeight});
+                };
+                return true;
+            }, {0});
+
         CrossBoundaryResult result;
-        ocean_index_block(result, b.write());
+        ocean_index_block(result, b.write(), pools);
```

Comment:
ideally inject json rpc client but `// Error: Syn(Error("async function is not directly supported yet, but see https://cxx.rs/async.html for a working approach,
// and https://github.com/pcwalton/cxx-async for some helpers; eventually what you wrote will work but it isn't integrated into the cxx::bridge macro yet"))`